prog/storage/migrate_spdk_vm_to_ubiblk.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+class Prog::Storage::MigrateSpdkVmToUbiblk < Prog::Base
+  subject_is :vm
+
+  def self.assemble(vm_id)
+    unless (vm = Vm[vm_id])
+      fail "Vm does not exist"
+    end
+
+    unless vm.vm_storage_volumes.count == 1
+      fail "This prog only supports Vms with exactly one disk"
+    end
+
+    unless vm.vm_storage_volumes.first.use_bdev_ubi
+      fail "Vm is already using Ubiblk"
+    end
```

Comment:
Checking for `if vm.vm_storage_volumes.first.vhost_block_backend_id` is a more direct way to check if the volume is using ubiblk.